src/main/ngapp/accio/src/app/editor/magic-wand.service.ts

```diff
@@ -0,0 +1,153 @@
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class MagicWandService {
+  /** Return set of coordinates(formatted as a 1-D array).
+   *  Coordinates correspond to pixels considered part of the mask.
+   */
+  /* tslint:disable */
+  floodfill(imgData: ImageData, xCoord: number, yCoord: number,
+            tolerance: number): Set<number> {
+    // Use basic queue to immitate recursive approach of the stack
+    // possible TODO: set array size after mvp to optimize heap memory access
+    const visit: Array<Array<number>> = new Array();
+    const visited: Set<number> = new Set();
+    // Use a set for mask; mainly do iter and set operations on masks
+    const mask: Set<number> = new Set();
```

Comment:
I assume this stores the index of the R pixel?

Reply:
visited stores the index of the start of the pixel representation in the DataArray, and yes the start of the pixel representation is the attribute for red of that pixel.